code/mob.dm

```diff
@@ -2658,9 +2658,13 @@
 			else
 				if (force_instead || alert(src, "Use the name [newname]?", newname, "Yes", "No") == "Yes")
 					if(!src.traitHolder.hasTrait("immigrant"))// stowaway entertainers shouldn't be on the manifest
-						var/datum/data/record/B = FindBankAccountByName(src.real_name)
-						if (B?.fields["name"])
-							B.fields["name"] = newname
+						for (var/L in list(data_core.bank, data_core.security, data_core.general, data_core.medical))
+							if (L)
+								var/datum/data/record/R = FindRecordByField(L, "name", src.real_name)
+								if (R)
+									R.fields["name"] = newname
+									if (R.fields["full_name"])
+										R.fields["full_name"] = newname
```

Comment:
I don't know much about the databank code, so I'm a bit confused here. Why do you null check the key "full_name" before assignment, but not "name"? What's the difference with these?

Reply:
The function `FindRecordByField(list, field, value)` either returns null, or a record with the field `field` with the value `value`. So if `R` is not null, then `R.fields[name]` is guaranteed to be not null. But I didn't check whether or not the record returned by `FindRecordByField` has the field `full_name`; I think only the security and general records have that field. So if the record has that field, we additionally update it. I don't think any more null checks are necessary, but if it would improve legibility I could restructure this to treat the two fields equally (at the expense of a couple more null checks when a player spawns).

Reply:
Gotchya, that makes sense. It feels kinda wonky, but that's not your fault. That's just how our databank code is considering they're all just assoc lists and not datums I suppose.